### PR TITLE
Dynamic bridge endpoint

### LINKED ISSUE #25: Operator equalsDID is not safe for every possible DID
Only feasible solution: make sure it is safe for all supported DID methods. The operator could also have a hard-coded did method whitelist.

It should be safe for the DIDs currently supported, but it needs to be checked.

### LINKED ISSUE #28: Add test cases with multiple VCs in one VP

### LINKED ISSUE #36: Add README section about new flow option fully on mobile phone

### PR DESCRIPTION
**Pull request by ilaydacansinkoc**

### CODE REVIEW

**Review by flhps — CHANGES_REQUESTED** [review]
Overall, lots of good changes. But there are a couple of things I am not super happy about. Would love to get some input on my comments @ilaydacansinkoc, but no code changes required. I will do whatever changes after hearing your thoughts.

File: README.md [hunk 1]
```diff
@@ -384,6 +450,138 @@ logical operators that can combine multiple constraints:
...
+      }
+    ]
+  }
+]
+```
+
+<!-- prettier-ignore -->
+> [!IMPORTANT] 
+> Each `credentialId` should be unique across all policy objects,
+> and should have integer string values starting from 1, incrementing by 1 for each subsequent policy object. This helps us determine
```
[the remark this answers is not in the document]
Comment (ilaydacansinkoc) [reply README.md:495]: **For the first comment:**

The `credentialId` helps to extract the correct claims from the correct policy.

I guess I added this when I tested a scenario when we want to extract same claims from the different type of VCs.
(2 VCs of different type but the field that we want to extract is the same: `$.credentialSubject.name`)

In practice, I think we can give unique string IDs again, but then we need to make sure that the constraints are evaluated correctly as well (`isValidConstraintFit`, and `resolveValue` functions). In these functions, I use `credentialId` to locate the correct VC to apply constraints. 

---

**For the second comment:**

It's required to be an integer-valued string. "1" or "2" for example.

File: README.md [hunk 2]
```diff
@@ -384,6 +450,138 @@ logical operators that can combine multiple constraints:
...
+> the correct policy object to apply to the VCs.
+
+<!-- prettier-ignore -->
+> [!IMPORTANT] 
+> Altough the `type` field is an optional parameter, it needs to be
```
[the remark this answers is not in the document]
Comment (ilaydacansinkoc) [reply README.md:500]: You might be right; I think this way: Each organization would have a certain type of VC, and the service provider would accept that type of VC.

The idea is similar to having registries so the verifier can know what credentials to accept.

But then we need to tell the code to apply the correct policy to the correct type of VC.

File: vclogin/config/loginPolicy.ts
[the remark this answers is not in the document]
Comment (ilaydacansinkoc) [reply vclogin/config/loginPolicy.ts:1]: Do you mean the async await use?

You might be right, but it caused some issues while performing the first login: It couldn't read the policy file when it supposed to, and therefore getting errors in the wallet app.

File: test_client.sh
[the remark this answers is not in the document]
Comment (ilaydacansinkoc) [reply test_client.sh:1]: Right, maybe adding a description to the readme on how to configure it instead would be a better solution.

File: vclogin/middleware.ts
[the remark this answers is not in the document]
Comment (ilaydacansinkoc) [reply vclogin/middleware.ts:1]: https://nextjs.org/docs/app/building-your-application/routing/middleware

This is basically a NextJS middleware that runs before a request is completed.
I am not sure if we can change the file name.

File: vclogin/lib/generatePresentationDefinition.ts [hunk 3]
```diff
@@ -40,29 +46,35 @@ export const generatePresentationDefinition = (policy: LoginPolicy) => {
       },
     },
     id: crypto.randomUUID(),
-    name: "SSI-to-OIDC Bridge",
+    name: "VC Login Service",
```
[the remark this answers is not in the document]
Comment (ilaydacansinkoc) [reply vclogin/lib/generatePresentationDefinition.ts:49]: I am not sure why I changed this. I guess you can just use the other one as well.

File: vclogin/lib/generatePresentationDefinition.ts [hunk 4]
```diff
@@ -40,29 +46,35 @@ export const generatePresentationDefinition = (policy: LoginPolicy) => {
...
+      "Using input descriptor override for incremental authorization",
+      pd,
+    );
+    return pd;
   }
 
   for (let expectation of policy) {
     if (expectation.patterns.length > 1) {
       let req = {
-        name: "Group " + expectation.credentialId,
```
[the remark this answers is not in the document]
Comment (ilaydacansinkoc) [reply vclogin/lib/generatePresentationDefinition.ts:56]: I guess I removed it mistakenly

File: vclogin/pages/api/clientMetadata.ts [hunk 5]
```diff
@@ -3,59 +3,21 @@
  * SPDX-License-Identifier: MIT
  */
 
+import { getMetadata } from "@/lib/getMetadata";
 import type { NextApiRequest, NextApiResponse } from "next";
-import { withLogging } from "@/middleware/logging";
```
[the remark this answers is not in the document]
Comment (ilaydacansinkoc) [reply vclogin/pages/api/clientMetadata.ts:7]: No 😀

File: vclogin/middleware.ts
[the remark this answers is not in the document]
Comment (jfelixh) [reply vclogin/middleware.ts:1]: Name is mandatory for built-in NextJS middleware system.

**Review by flhps — APPROVED** [review]

**Comment by jdsika** [comment]
Why is the build not triggered? Can you confirm the build locally?

**Comment by jfelixh** [comment]
> Why is the build not triggered? Can you confirm the build locally?

I have no idea. The normal build on Node 20.x works (locally and here). But the old built on 18.x, that we removed from the github action, is still showing up here. It was already removed in the last merged PR, since 18.x is pretty old and it should be sufficient to have the tests and build run on one Node version.

**Review by jdsika — APPROVED** [review]

**Pull request merged**